src/dotnet/MsbuildProject.cs

```diff
@@ -166,12 +174,97 @@ public static void EnsureAllReferencesExist(List<string> references)
             }
         }
 
+        public IEnumerable<NuGetFramework> GetTargetFrameworks()
```

Comment:
Since I've removed GetTargetFrameworks implementation from here, I don't believe it is too bad now. I'd prefer to move From.* methods to some other class but I believe that any .*Factory class makes code less readable as you need to know about its existence to use it. As we talked offline with Peter we both agree that static extension methods would be perfect solution here but Factory IMO is not a good idea.